fuzz-test/readme.md

```diff
@@ -0,0 +1,20 @@
+# Fuzz test
+Simple setup using AFL or libFuzzer.
+
+## Libfuzzer
+```
+$ mkdir build-libfuzzer
+$ cd build-libfuzzer
+$ cmake ../.. -DCMAKE_C_COMPILER=clang -DCMAKE_C_FLAGS="-fsanitize=fuzzer,address -DLIBFUZZER" -DCMAKE_C_COMPILER_WORKS=On
+$ make fuzz_pet
+$ ./fuzz-test/fuzz_pet
+```
+
+## AFL
+```
+$ mkdir build-afl
+$ cd build-afl
+$ export AFL_USE_ASAN=1
+$ cmake ../.. -DCMAKE_C_COMPILER=afl-clang-fast
+$ make fuzz_pet
```

Comment:
Missing execution instructions here.

Reply:
Hehe yes. AFL also requires some initial feed, should add that when i have time. It is also possible to look in AFL documentation

Reply:
Yes, I tried your PR by making a dir `dir_in` with a file `empty.dict` with simply:

```txt
"0"
```

Just in order to satisfy AFL's input requirement.

I'm not sure if there is anything particularly clever to put into the `.dict` file since the input to the program is binary.

Reply:
I normally do something like that also. Another approach is to use libFuzzer, that doesn't require an initial seed. I've been looking for some good cbor dictionary for fuzzing but haven't found any yet. In another project (similar to cbor), this dict was used: https://github.com/assaabloy-ppi/binson-c-light/blob/master/fuzz-test/binson.dict . Something similar to that but for cbor could help.

I created this PR to initiate a discussion on fuzzing strategy, i don't know what is the best way to go. I think something nice in the end would to use oss-fuzz (https://github.com/google/oss-fuzz). But a good start is being able to run it on your machine locally.

I think something that fuzzes:
- Functions from cbor_decode.c directly
- Generated parsing function based on some cddl input. Just try to figure out what kind of CDDL structure gives the best coverage

Reply:
`AFL_USE_ASAN=1` would not work on my setup. Maybe it works if I build it as a 32 bit binary. @sijohans what flags did you use to build for 32 bit?